kedro/framework/cli/starters.py

```diff
@@ -149,11 +241,30 @@ def starter():
 @starter.command("list")
 def list_starters():
     """List all official project starters available."""
-    repo_url = _STARTERS_REPO.replace("git+", "").replace(".git", "/tree/main/{alias}")
-    output = [
-        {alias: repo_url.format(alias=alias)} for alias in sorted(_STARTER_ALIASES)
-    ]
-    click.echo(yaml.safe_dump(output))
+    starters_aliases = _get_starters_aliases()
+    starters_aliases_by_origin = {
+        # origin: [
+        #     {
+        #         k: v
+        #         for k, v in config.items()
+        #         if k in {"name", "template_path", "directory"} and v is not None
+        #     }
+        #     for config in config_list
+        # ]
+        origin: list(configs)
+        for origin, configs in groupby(
+            sorted(starters_aliases, key=itemgetter("origin")), key=itemgetter("origin")
+        )
+    }
+
+    # ensure kedro starters are listed first
+    kedro_starters = starters_aliases_by_origin.pop("kedro")
+    click.secho(f"\nBuilt-in starters\n", fg="yellow")
+    click.echo(yaml.dump(kedro_starters))
+
+    for origin, starter_config in starters_aliases_by_origin.items():
+        click.secho(f"\nStarters from {origin}\n", fg="yellow")
+        click.echo(yaml.dump(starter_config))
```

Comment:
After spending WAAAY too long playing around with this (and for some time unknowingly working on something that just reproduced what you had obviously thought of before and commented out...), here's the best I've come up with:
```
    starters_aliases_by_origin = groupby(
        sorted(_get_starters_aliases(), key=itemgetter("origin")),
        key=itemgetter("origin"),
    )
    for origin, starter_config in starters_aliases_by_origin:
        click.secho(f"\nStarters from {origin}\n", fg="yellow")
        for starter in starter_config:
            del starter["origin"]
            name = starter.pop("name")
            click.echo(yaml.dump({name: starter}))
```
![image](https://user-images.githubusercontent.com/49395058/157484352-bceed0cc-5157-4cfe-9e30-4af5c25f3fb5.png)

If we care about ordering so that `kedro` is always first, you could do the below, but personally I don't think it's worth the extra complexity and would just leave it unordered.
```
    starters_aliases_by_origin = {k: list(v) for k, v in groupby(
        sorted(_get_starters_aliases(), key=itemgetter("origin")),
        key=itemgetter("origin"),
    )}
    for origin, starter_config in sorted(starters_aliases_by_origin.items(), key=lambda x: x[0] != "kedro"):
```

Reply:
Yes I played a lot with the formatting too and get to the same conclusion as you. I'd personnally prefer showing built-in starters first but it is really not a big deal.